### PR TITLE
chore(@automattic/composite-checkout): Update webpack-dev-server

### PR DESCRIPTION
**Pull request by scinos**
#### Changes proposed in this Pull Request

* Uses a version of webpack-dev-server compatible with webpack 5

[Changelog](https://github.com/webpack/webpack-dev-server/blob/master/CHANGELOG.md)

#### Testing instructions

* Checkout this branch and run `composite-checkout-demo`. Verify `http://localhost:3000` loads without problems.

### CODE REVIEW

**Comment by scinos** [comment]
Moved to [draft] until I understand why the increase in bundle size

**Review by sirbrillig — COMMENTED** [review]
This looks fine in general. Let me know if I can help!

**Review by tyxla — COMMENTED** [review]
Seems to need a rebase. Also, any finds as to where the bundle changes came from?

**Comment by scinos** [comment]
> Also, any finds as to where the bundle changes came from?

Yes! And it is rather interesting:

* Before webpack 5 - Webpack 4 uses its own set of polyfills for node modules. So when a package does `require("util")`, webpack bundled its own polyfill. As `util` is a node module, many packages don't have an explicit dependency on `util` in their `package.json`.

* With webpack 5 (i.e. `trunk`) - Webpack doesn't include those polyfills anymore, is up to the consumer to install them. Because we have some packages depending on `util` (from https://github.com/browserify/node-util), we end up having `util@0.11.1` in `node_modules/util`, and that's the version webpack will use to polyfill `require("util")`. 

* With this PR - `webpack-dev-server` brings in a new `util` version that gets hoisted, and now `node_modules/util` points to `util@0.12.3`. That's the version webpack will use to polyfill `require("util")`. As such, any bundle that were doing `require("util")` will now use a different version of `util`. This new version of `util` is bigger because it has more dependencies, therefore the bundlers are bigger.

---

I think the new situation is correct: webpack uses the latest `util` to polyfill `require('util')`. But I don't like that it happens by accident: if `yarn` decides to hoist an older version of `util`, bundle sizes will change again for no apparent reason.

Probably the solution (as hinted in https://github.com/browserify/node-util/issues/57) is to have a explicit alias in webpack config, making sure `util` is always resolved to a specific version and not to whatever happens to get hoisted.

/cc @jsnajdr

**Comment by jsnajdr** [comment]
> have a explicit alias in webpack config, making sure `util` is always resolved to a specific version

Yes, we already have a similar alias for some `@wordpress/*` packages (see also #48162).

But where does that `require( 'util' )` come from? I can't find any `util` import in the Calypso codebase itself. So is it some third party package?

Using Node packages in browser code is undesired and should be eliminated if possible.

**Comment by scinos** [comment]
> But where does that require( 'util' ) come from? I can't find any util import in the Calypso codebase itself. So is it some third party package?

The one I found comes via `./state/utils/schema-utils.js` -> `node_modules/is-my-json-valid/index.js` -> `node_modules/generate-function/index.js` -> `node_modules/util/util.js`

**Comment by jsnajdr** [comment]
@scinos and `npm run whybundled` tells me that's actually the only place where the `util` module is used:
```
 MODULE  util
├─ imported: 1 time
├─ deps count: 1
├─ size: 20 KiB [for all included files]
├─ type: [transitive]
│  └─ util -> generate-function -> is-my-json-valid
│  
├─ chunks: vendors-node_modules_uuid_dist_esm-browser_v4_js-node_modules_crc32_lib_crc32_js-node_modules-26c5b7
├─ locations: 
│  └─ ../node_modules/util/
│  
├─ files: 
│  ├─ ../node_modules/util/support/isBufferBrowser.js
│  └─ ../node_modules/util/util.js
│  
└─ reasons:
   └─ generate-function
      └─ ../node_modules/generate-function/index.js  1:11-26  [cjs require]
```

**Comment by scinos** [comment]
Update: A different PR was created and merged to fix the version of `util` (#50074). As expected, that PR increased bundle size. Now that it is in trunk, this PR doesn't change bundle size in a significant way anymore.

File: yarn.lock [hunk 1]
```diff
@@ -12932,20 +12943,20 @@ fault@^1.0.0, fault@^1.0.1:
...
-
-faye-websocket@~0.11.1:
+faye-websocket@^0.11.3:
   version "0.11.3"
   resolved "https://registry.yarnpkg.com/faye-websocket/-/faye-websocket-0.11.3.tgz#5c0e9a8968e8912c286639fde977a8b209f2508e"
   integrity sha512-D2y4bovYpzziGgbHYtGCMjlJM36vAl/y+xUyn1C+FVx8szd1E+86KwVw6XvYSzOP8iMpm1X0I4xJD+QtUb36OA==
   dependencies:
     websocket-driver ">=0.5.1"
 
+faye-websocket@~0.10.0:
```
Comment (tyxla) [inline yarn.lock:12953]: Could this one use some deduping?

File: yarn.lock [hunk 2]
```diff
@@ -13606,7 +13617,12 @@ fsevents@^1.2.11, fsevents@^1.2.7:
     bindings "^1.5.0"
     nan "^2.12.1"
 
-fsevents@^2.1.2, fsevents@~2.1.2:
+fsevents@^2.1.2, fsevents@~2.3.1:
+  version "2.3.1"
+  resolved "https://registry.yarnpkg.com/fsevents/-/fsevents-2.3.1.tgz#b209ab14c61012636c8863507edf7fb68cc54e9f"
+  integrity sha512-YR47Eg4hChJGAB1O3yEAOkGO+rlzutoICGqGo9EZ4lKWokzZRSyIW1QmTzqjtw8MJdj9srP869CuWw/hyzSiBw==
+
+fsevents@~2.1.2:
```
Comment (tyxla) [inline yarn.lock:13625]: Seems like this could use some deduping?

**Review by tyxla — APPROVED** [review]
There might be some candidates for deduping, but can be done in subsequent PRs. This looks good and works as expected 🚢

File: yarn.lock (hunk 1 above)
Comment (scinos) [reply yarn.lock:12953]: Thanks for checking :) In this case, `~0.10.0` and `^0.11.3` don't overlap. 

I've found https://semver.npmjs.com/ really useful to verify what versions can satisfy a range.

File: yarn.lock (hunk 2 above)
Comment (scinos) [reply yarn.lock:13625]: Same, `~2.1.2` and `~2.3.1` (or `^2.1.2`) don't overlap

**Pull request merged**